app/Controllers/subscriptionController.php

```diff
@@ -87,17 +87,23 @@ public function feedAction() {
 
 			$cat = intval(Minz_Request::param('category', 0));
 
+			$mute = Minz_Request::param('mute', false);
+			$ttl = intval(Minz_Request::param('ttl', FreshRSS_Feed::TTL_DEFAULT));
+			if ($mute && FreshRSS_Feed::TTL_DEFAULT === $ttl) {
+				$ttl = FreshRSS_Context::$user_conf->ttl_default;
+			}
+
 			$values = array(
 				'name' => Minz_Request::param('name', ''),
 				'description' => sanitizeHTML(Minz_Request::param('description', '', true)),
 				'website' => checkUrl(Minz_Request::param('website', '')),
 				'url' => checkUrl(Minz_Request::param('url', '')),
 				'category' => $cat,
 				'pathEntries' => Minz_Request::param('path_entries', ''),
-				'priority' => intval(Minz_Request::param('priority', 0)),
+				'priority' => intval(Minz_Request::param('priority', FreshRSS_Feed::PRIORITY_MAIN_STREAM)),
 				'httpAuth' => $httpAuth,
 				'keep_history' => intval(Minz_Request::param('keep_history', -2)),
-				'ttl' => intval(Minz_Request::param('ttl', -2)),
+				'ttl' => $ttl * ($mute ? -1 : 1),
```

Comment:
Does that work for default TTL values (0)?

Reply:
yes it does. The code force the use of the default TTL when used with default. The only drawback is that you need to set it back to default when you unmute